specification.markdown

```diff
@@ -245,6 +245,18 @@ A `Binary Element` MUST declare a length in octets from zero to `VINTMAX`.
 
 The contents of a `Binary Element` should not be interpreted by the `EBML Reader`.
 
+# Terminating Elements
+
+`Null Octets`, which are octets with all bits set to zero, MAY follow the value of a `String Element` or `UTF-8 Element` to serve as a terminator. An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets` in order to overwrite a stored value with a new value of lesser length while maintaining the same `Element Data Size` (this can prevent the need to rewrite large portions of an `EBML Document`); otherwise the use of `Null Octets` within a `String Element` or `UTF-8 Element` is NOT RECOMMENDED. An `EBML Reader` MUST consider the value of the `String Element` or `UTF-8 Element` to be terminated upon the first read `Null Octet` and ignore any following data within that `Element`. If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`. A string value and a copy of that string value terminated by one or more `Null Octets` are semantically equal.
```

Comment:
About "this can prevent the need to rewrite large portions of an `EBML Document`" and the whole sentence. I don't like that part because it promotes the wrong method of overwriting elements. We already have a perfectly valid element for such a use case: `EbmlVoid`. As the `ElementID` of `EbmlVoid` is only a single byte, the shortest-possible ID, you can use an `EbmlVoid` to overwrite any element in any situation without having to re-write or overwrite anything else.

I'm OK with allowing such termination, but I'd rather we don't promote it. I'd vote for ending the sentence much ealier, e.g.:

> An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets`, but it is NOT RECOMMENDED.

We could even state that instead of terminating an existing element in order to make it empty one should use an `EbmlVoid`.